contracts/core/AMM/SportsAMMV2.sol

```diff
@@ -1048,77 +1005,78 @@ contract SportsAMMV2 is Initializable, ProxyOwned, ProxyPausable, ProxyReentranc
+        safeBoxFee = _safeBoxFee;
+        emit AmountsUpdated(_safeBoxFee);
     }
 
-    /// @notice sets additional payout percentage for certain collaterals
-    /// @param _collateral to add extra payout for
-    /// @param _addedPayout percentage amount for extra payout
-    function setAddedPayoutPercentagePerCollateral(address _collateral, uint _addedPayout) external onlyOwner {
+    /// @notice Sets parameters related to a collateral asset.
+    /// @dev Only updates fields if the new value differs from the stored one.
+    /// @param _collateral The collateral token address.
+    /// @param _liquidityPool New liquidity pool address (optional, skip if same).
+    /// @param _addedPayout New added payout percentage (optional, skip if same).
+    /// @param _safeBox New dedicated SafeBox address (optional, skip if same).
+    function configureCollateral(
+        address _collateral,
+        address _liquidityPool,
+        uint _addedPayout,
+        address _safeBox
+    ) external onlyOwner {
+        // Liquidity pool update
+        if (liquidityPoolForCollateral[_collateral] != _liquidityPool) {
+            _updateApproval(IERC20(_collateral), liquidityPoolForCollateral[_collateral], _liquidityPool);
+            liquidityPoolForCollateral[_collateral] = _liquidityPool;
+        }
+
+        // Added payout percentage update
         addedPayoutPercentagePerCollateral[_collateral] = _addedPayout;
-        emit SetAddedPayoutPercentagePerCollateral(_collateral, _addedPayout);
-    }
 
-    /// @notice sets dedicated SafeBox per collateral
-    /// @param _collateral to set dedicated SafeBox for
-    /// @param _safeBox for the given collateral
-    function setSafeBoxPerCollateral(address _collateral, address _safeBox) external onlyOwner {
+        // SafeBox override update
         safeBoxPerCollateral[_collateral] = _safeBox;
-        emit SetSafeBoxPerCollateral(_collateral, _safeBox);
+
+        emit CollateralConfigured(_collateral, _liquidityPool, _addedPayout, _safeBox);
+    }
+
+    function _updateApproval(IERC20 token, address oldSpender, address newSpender) internal {
+        if (oldSpender != address(0)) {
+            token.approve(oldSpender, 0);
+        }
+
+        if (newSpender != address(0)) {
+            token.approve(newSpender, MAX_APPROVAL);
+        }
     }
 
     /* ========== MODIFIERS ========== */
 
     modifier onlyKnownTickets(address _ticket) {
-        require(manager.isKnownTicket(_ticket), "UnknownTicket");
+        if (!manager.isKnownTicket(_ticket)) revert UnknownTicket();
         _;
     }
 
     modifier onlyWhitelistedAddresses() {
-        require(
-            manager.isWhitelistedAddress(msg.sender, ISportsAMMV2Manager.Role.ROOT_SETTING) || msg.sender == owner,
-            "InvalidSender"
-        );
+        if (!(manager.isWhitelistedAddress(msg.sender, ISportsAMMV2Manager.Role.ROOT_SETTING) || msg.sender == owner))
```

Comment:
`!manager.isWhitelistedAddress(msg.sender, ISportsAMMV2Manager.Role.ROOT_SETTING) && msg.sender != owner` - move negation inside, just to be consistent

Reply:
done